sunpy/instr/fermi.py

```diff
@@ -0,0 +1,373 @@
+    scz : array
+        Two-element array containing the RA/DEC information of the Fermi spacecraft Z-axis
+    time : datetime object
+        The time corresponding to the input scx and scz values, in a format understandable by parse_time.
+
+    Returns
+    -------
+    dict
+        A dictionary containing the RA/DEC for each Fermi/GBM NaI detector at the given input time.
+    '''
+
+    scx_vector = (np.array([np.cos(scx[0].to('rad').value)*np.cos(scx[1].to('rad').value), 
+                        np.sin(scx[0].to('rad').value)*np.cos(scx[1].to('rad').value),
+                        np.sin(scx[1].to('rad').value)]))
+
+    scz_vector = (np.array([np.cos(scz[0].to('rad').value)*np.cos(scz[1].to('rad').value), 
+                        np.sin(scz[0].to('rad').value)*np.cos(scz[1].to('rad').value),
+                        np.sin(scz[1].to('rad').value)]))
+
+    # for each detector, do the rotation depending on the detector zenith and azimuth angles
+    detector_radecs = copy.deepcopy(detectors)
+    for l, d in detectors.items():
+        phi = d[0].value
+        theta = d[1].value
+
+        # rotate about spacecraft z-axis first
+        vx_primed = rotate_vector(scx_vector, scz_vector, np.deg2rad(phi))
+
+        # now find spacecraft y-axis using cross product
+        vy_primed = np.cross(scz_vector, vx_primed)
+
+        # do the second part of the rotation around vy
+        vz_primed = rotate_vector(scz_vector, vy_primed, np.deg2rad(theta))
+
+        # now we should be pointing at the new RA/DEC.
+        ra = Longitude(np.degrees(math.atan2(vz_primed[1], vz_primed[0])) * u.deg)
+        dec = Latitude(np.degrees(math.asin(vz_primed[2])) * u.deg)
+
+        # save the RA/DEC in a dictionary
+        detector_radecs[l] = [ra, dec]
+
+    detector_radecs['time'] = time
+    return detector_radecs
+
+
+def rotate_vector(vector, axis, theta):
+    # the Euler-Rodrigues formula for rotating vectors
+    # axis is the vector to rotate around
+    # theta is the angle to rotate
+    # http://en.wikipedia.org/wiki/Euler-Rodrigues_parameters#Rotation_angle_and_rotation_axis
+
+    axis = axis/np.sqrt(np.dot(axis, axis))
+    a = np.cos(theta/2)
+    b, c, d = -axis*np.sin(theta/2)
+
+    rot_matrix = np.array([[a*a+b*b-c*c-d*d, 2*(b*c+a*d),2*(b*d-a*c)],
+                     [2*(b*c-a*d), a*a+c*c-b*b-d*d, 2*(c*d+a*b)],
+                     [2*(b*d+a*c), 2*(c*d-a*b), a*a+d*d-b*b-c*c]])      
+
+    return np.dot(rot_matrix,vector)
+
+
+def get_detector_separation_angles(detector_radecs, sunpos):
+    '''Finds the separation angle between the Sun and each NaI detector, given a dictonary of detector RA/DECs.'''
+    angles = copy.deepcopy(detector_radecs)
+    for l, d in detector_radecs.items():
+        if not l == 'time':
+            angle = separation_angle(d, sunpos)
+            angles[l] = angle
+    return angles
+
+
+def separation_angle(radec1, radec2):
```

Comment:
You could use `astropy.coordinates` to handle your on sky coordinates correctly, and there is a seperation method on `SkyCoord` if I remember.